File: scripts/migrate_titles.py
import json
import os

recipes_file = 'database/recipes.json'

if not os.path.exists(recipes_file):
    print(f"Error: {recipes_file} not found")
    exit(1)

with open(recipes_file, 'r', encoding='utf-8') as f:
    recipes = json.load(f)

modified_count = 0
for recipe in recipes:
    if 'translations' in recipe:
        for lang in recipe['translations']:
            if 'title' in recipe['translations'][lang]:
                del recipe['translations'][lang]['title']
                modified_count += 1

if modified_count > 0:
    with open(recipes_file, 'w', encoding='utf-8') as f:
        json.dump(recipes, f, indent=2, ensure_ascii=False)
    print(f"Migration complete. Removed titles from {modified_count} translations.")
else:
    print("No translated titles found to remove.")

File: scripts/migrate_titles.js
const fs = require('fs');
const path = require('path');

const recipesFile = path.join(__dirname, '..', 'database', 'recipes.json');

function migrate() {
    if (!fs.existsSync(recipesFile)) {
        console.error('Recipes file not found at:', recipesFile);
        return;
    }

    try {
        const data = fs.readFileSync(recipesFile, 'utf-8');
        const recipes = JSON.parse(data);

        let modifiedCount = 0;
        recipes.forEach(recipe => {
            if (recipe.translations) {
                Object.keys(recipe.translations).forEach(lang => {
                    if (recipe.translations[lang].title !== undefined) {
                        delete recipe.translations[lang].title;
                        modifiedCount++;
                    }
                });
            }
        });

        if (modifiedCount > 0) {
            fs.writeFileSync(recipesFile, JSON.stringify(recipes, null, 2));
            console.log(`Migration complete. Removed titles from ${modifiedCount} translations.`);
        } else {
            console.log('No translated titles found to remove.');
        }
    } catch (err) {
        console.error('Error during migration:', err);
    }
}

migrate();
